metis-indexing/src/main/java/eu/europeana/indexing/FullBeanPublisher.java

```diff
@@ -159,4 +189,68 @@ private void publishToSolr(RdfWrapper rdf, FullBeanImpl fullBean) throws Indexin
       throw new RecordRelatedIndexingException(SOLR_SERVER_PUBLISH_ERROR, e);
     }
   }
+
+  private String searchMatchingRecordForRedirection(String recordIdIncludingDatasetId,
+      List<String> datasetIdsForRedirection)
+      throws IndexerRelatedIndexingException, RecordRelatedIndexingException {
+    //Check if record with same record identifier exists
```

Comment:
I think this code skips redirects if the record already exists? Because we assume that the redirects would have already been added during the previous indexing? Is this an assumption we can make? What if the user made a mistake and changed the previous dataset list and wishes to index again?
I think we should either add comments to highlight this assumption, or, even better, skip this step and recompute the redirects.

Reply:
Yes that is the assumption. There reasoning here is improvement in performance

Reply:
Created a ticket for this to have a bit more time to think about this operation